pkgs/top-level/python-packages.nix

```diff
@@ -13901,6 +13901,10 @@ self: super: with self; {
 
   rubymarshal = callPackage ../development/python-modules/rubymarshal { };
 
+  ruff = toPythonModule (pkgs.ruff.override {
```

Comment:
I don't think we need the toPythonModule here if we are already using buildPythonPackage

Reply:
Doesn't that bring `python311Packages.ruff` which might be needed by python3.11 dependencies?

Reply:
Yes, this is required to build a python module that depends on ruff.

See `python311Packages.pytest-ruff` in https://github.com/NixOS/nixpkgs/pull/350635